ext/opcache/jit/zend_jit_ir.c

```diff
@@ -2688,6 +2746,23 @@ static void zend_jit_init_ctx(zend_jit_ctx *jit, uint32_t flags)
 			 */
 			if (GCC_GLOBAL_REGS) {
 				jit->ctx.fixed_save_regset = IR_REGSET_PRESERVED & ~((1<<ZREG_FP) | (1<<ZREG_IP));
+			} else if (ZEND_VM_TAIL_CALL_DISPATCH) {
+				/* The only preserved register on x86 is RBP:
+				 * https://github.com/llvm/llvm-project/blob/a414877a7a5f000d01370acb1162eb1dea87f48c/llvm/lib/Target/X86/X86RegisterInfo.cpp#L319
+				 * https://github.com/llvm/llvm-project/blob/68bfe91b5a34f80dbcc4f0a7fa5d7aa1cdf959c2/llvm/lib/Target/X86/X86CallingConv.td#L1183
+				 * On AArch64 it's LR, FP:
+				 * https://github.com/llvm/llvm-project/blob/68bfe91b5a34f80dbcc4f0a7fa5d7aa1cdf959c2/llvm/lib/Target/AArch64/AArch64CallingConvention.td#L681
+				 *
+				 * Add them to the fixed_regset to prevent usage or these regs.
+				 * It's cheaper to not use them than to save them.
+				 */
+#if defined(IR_TARGET_X64)
+				jit->ctx.fixed_regset |= (1<<IR_REG_FP);
+#elif defined(IR_TARGET_AARCH64)
+				jit->ctx.fixed_regset |= (1<<IR_REG_FP) | (1<<IR_REG_LR);
+#else
+				ZEND_UNREACHABLE();
+#endif
```

Comment:
``jit->ctx.fixed_stack_frame_size`` assigned above may be reduced as we don't need space to save/restore any register now.